src/scripts/main.js

```diff
@@ -1,3 +1,59 @@
 'use strict';
 
 // write code here
+
+const employees = document.querySelectorAll('li');
+const listEmployees = document.querySelector('ul');
+
+function getSalary(element) {
+  const salary = element.dataset.salary;
+
+  if (!salary) {
+    throw new Error(`Salary not found for employee: ${element.textContent}`);
+  }
+
+  const cleanedSalary = salary.substring(1).replace(',', '');
+  const numericSalary = Number(cleanedSalary);
+
+  if (isNaN(numericSalary)) {
+    throw new Error(`Invalid salary format for employee:
+      ${element.textContent}`);
+  }
+
+  return numericSalary;
+}
+
+function sortList(list) {
+  const listItem = Array.from(list);
+
+  const sortedList = listItem.slice().sort((a, b) => {
+    return getSalary(b) - getSalary(a);
+  });
+
+  listEmployees.innerHTML = '';
+
+  sortedList.forEach((element) => {
+    listEmployees.appendChild(element);
+  });
```

Comment:
Appending child nodes inside a loop can cause performance issues because each operation causes the page to re-render. Instead, you can create a document fragment, append your items to that, and then append the fragment to the DOM. This way, the page only re-renders once.